pkg/usrlocalsharelima/usrlocalsharelima.go

```diff
@@ -16,8 +16,11 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// executable is a variable that can be overridden in tests.
+var executable = os.Executable
```

Comment:
I would just cache the return value of the function because it is constant over the lifetime of the process. Import `Must` at the top:

```
import (
    …
    . "github.com/lima-vm/lima/pkg/must"
)
```

and then use

```suggestion
var self = Must(os.Executable())
```

later you can simply assign to it:

```go
    self = limactlPath
```

See `osutil/user.go` for an example of using `Must()`.

Reply:
Sir @jandubois ,
I have tried implementing the above suggestions, I need your suggestion with the failing tests.